### PR TITLE
Improve: Add reproducible random seed support for ML model training

### LINKED ISSUE #2370: Improve: Add Reproducible Random Seed Support for ML Model Training
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What would you like to improve?

The current ML training pipeline relies on random dataset generation and model initialization, but no fixed random seed is configured.

In `backend/ml/train_scorer.py`, randomness is used extensively while generating synthetic training data:

```python
import random
import numpy as np

...

for _ in range(n_samples):
    prompt = random.choice(PROMPT_SEEDS)
    quality = random.choices(
        ["high", "medium", "low"],
        weights=[0.3, 0.5, 0.2]
    )[0]

    story = random_story(prompt, quality)
```

Since no random seed is set, each training run can generate different datasets, model weights, and evaluation results.

### Proposed Improvement

Add reproducible random seed support for all randomness sources used during training:

```python
import random
import numpy as np
import tensorflow as tf

SEED = 42

random.seed(SEED)
np.random.seed(SEED)
tf.random.set_seed(SEED)
```

Optionally, the seed value could be made configurable through environment variables or command-line arguments.

### Why does this matter?

Reproducibility is an important aspect of machine learning workflows.

Benefits include:

* Consistent results across multiple training runs
* Easier debugging and testing
* Reliable model performance comparisons
* Better collaboration among contributors
* Improved experiment tracking and validation

Currently, contributors may observe different outputs even when using the same training configuration, making it difficult to compare changes or investigate model behavior. Adding seed support would improve the reliability and maintainability of the ML training pipeline.

### Screenshots or screen recording (if applicable)

Not applicable.

Relevant file:
`backend/ml/train_scorer.py`

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

### PR DESCRIPTION
**Pull request by mirunalini277**
## Summary

This PR improves the reproducibility of the ML training pipeline by adding a fixed random seed configuration.

## Changes Made

* Added `import tensorflow as tf`
* Added a fixed seed value (`SEED = 42`)
* Added `random.seed(SEED)`
* Added `np.random.seed(SEED)`
* Added `tf.random.set_seed(SEED)`
* Added comments explaining the purpose of the seed configuration

## Why This Change?

The training script relies on random dataset generation and model initialization. Without a fixed random seed, training results may vary between runs, making debugging and model performance comparisons difficult.

This change helps ensure more consistent and reproducible training outcomes.

## Related Issue

Closes #2370

### CODE REVIEW

**Comment by mirunalini277** [comment]
Contributing under GSSoC'26 🚀

**Comment by ronisarkarexe** [comment]
pls give a star
